pkg/kubectl/cmd/get.go

```diff
@@ -316,12 +316,9 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 			return err
 		}
 
-		for ix := range infos {
-			objs[ix], err = infos[ix].Mapping.ConvertToVersion(infos[ix].Object, version)
-			if err != nil {
-				allErrs = append(allErrs, err)
-				continue
-			}
+		objs, err = resource.AsVersionedObjects(infos, version, f.JSONEncoder())
```

Comment:
yes, we are still passing `v1`, we will try to convert to `v1` version, but if failed, we can still use object's mapping version.